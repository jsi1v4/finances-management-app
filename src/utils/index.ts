import { createElement } from 'react';

/**
 * Wrap up Components
 * @param JSX.Element[] sequence of elements
 * @return wrapper (first element as father and others as children)
 */
export function wrapse(...comps) {
  return comps.reduceRight((wrap, comp) => ({ children }) =>
    createElement(wrap, null, createElement(comp, null, children))
  );
}

export default {
  wrapse
};
