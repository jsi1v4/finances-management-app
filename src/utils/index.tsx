import React from 'react';

/**
 * Wrap up Components
 * @param JSX.Element[] sequence of elements
 * @return wrapper (first element as father and others as children)
 */
export function wrapse(...components) {
  return components.reduceRight(
    (Wrapper, Comp) => ({ children }) => (
      <Wrapper>
        <Comp>{children}</Comp>
      </Wrapper>
    ),
    ({ children }) => <>{children}</>
  );
}

export default {
  wrapse
};
