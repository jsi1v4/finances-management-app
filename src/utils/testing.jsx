import React from 'react';

import { LocaleConfig } from '../i18n';

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

export function LocaleTesting({ children }) {
  return <LocaleConfig lang="en">{children}</LocaleConfig>;
}

export default LocaleTesting;
