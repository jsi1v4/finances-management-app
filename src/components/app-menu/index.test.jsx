import React from 'react';
import { render } from '@testing-library/react';

import { LocaleTesting } from '../../utils/testing';
import { AppMenu } from './index';

describe('AppMenu component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<AppMenu />, {
      wrapper: LocaleTesting,
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render component', async () => {
    const { getByTestId, getByText } = render(<AppMenu />, {
      wrapper: LocaleTesting,
    });
    expect(getByTestId('menu').innerHTML).toBeDefined();
    expect(getByText('Home').innerHTML).toBeDefined();
  });
});
