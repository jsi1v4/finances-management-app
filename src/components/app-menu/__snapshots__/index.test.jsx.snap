// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMenu component matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
    data-testid="menu"
    role="menu"
  >
    <li
      class="ant-menu-item ant-menu-item-only-child ant-menu-item-selected"
      role="menuitem"
      style="padding-left: 24px;"
    >
      Home
    </li>
  </ul>
</DocumentFragment>
`;
